widget/button.go

```diff
@@ -18,19 +18,29 @@ type buttonRenderer struct {
 	button  *Button
 }
 
+func (b *buttonRenderer) padding() fyne.Size {
+	if b.button.Text != "" {
+		return fyne.NewSize(theme.Padding()*4, theme.Padding()*2)
+	} else if b.icon != nil {
+		return fyne.NewSize(theme.Padding()*2, theme.Padding()*2)
+	}
+	return fyne.NewSize(0, 0)
```

Comment:
This seems like a fallback that should never occur - a button with neither text nor icon...
Wouldn't it be better to have either the text version or the non-text version be the special case then return the default otherwise? If the button was truly empty then with 0 padding it may be invisible...